test/systemtests/network_test.go

```diff
@@ -338,6 +338,7 @@ func (s *systemtestSuite) testNetworkAddDeleteTenant(c *C, encap, fwdmode string
 			}
 		}
 
+		time.Sleep(3 * time.Second)
```

Comment:
The timing issue is the delay from issuing the command from the system test code to the command taking effect. Because it is vagrant, sometimes it is slow to propagate.